src/Authentication/Authentication/test/Find-MgGraphPermission.Tests.ps1

```diff
@@ -2,15 +2,170 @@
+        }
+
+        It 'Executes successfully with no parameters' {
+            { Find-MgGraphPermission | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+
+        It "Retrieves the expected set of delegated and app-only permissions when a search string is specified" {
+            { Find-MgGraphPermission mail | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+
+        It "Returns nothing and throws no exception if a search string is specified and there is no match" {
+            { Find-MgGraphPermission 'Microsoft is the greatest company' | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+    }
+
+    Context "When executing the command without a connection to MS Graph" {
+        BeforeAll {
+            Mock Invoke-MgGraphRequest {
+
+                Throw [System.Management.Automation.ValidationMetadataException]::new('mock connection error message')
+
+            }
+        }
+
+        It 'Executes successfully with no parameters' {
+            { Find-MgGraphPermission | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+
+        It "Retrieves the expected set of delegated and app-only permissions when a search string is specified" {
+            { Find-MgGraphPermission mail | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+
+        It "Returns nothing and throws no exception if a search string is specified and there is no match" {
+            { Find-MgGraphPermission 'Microsoft is the greatest company' | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+        }
+    }
+
+    Context "When executing the command after initially not having a connection to MS Graph but getting it later" {
+
+        BeforeEach {
+            Mock Invoke-MgGraphRequest {
+
+                Throw [System.Management.Automation.ValidationMetadataException]::new('mock connection error message')
+
+            }
+        }
+
         It 'Executes successfully with no parameters' {
             { Find-MgGraphPermission | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+
+            Mock Invoke-MgGraphRequest{
+                $permissionData
+            }
+
+            { Find-MgGraphPermission | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+
+        }
+
+        It "Retrieves the expected set of delegated and app-only permissions when a search string is specified" {
+            { Find-MgGraphPermission mail | Out-Null } | Should -Not -Throw
+            Assert-MockCalled Invoke-MgGraphRequest
+
+            Mock Invoke-MgGraphRequest{
+                $permissionData
+            }
+
+            { Find-MgGraphPermission mail | Out-Null } | Should -Not -Throw
```

Comment:
Now you need to do more than just check the data. You must also check the results from the test file. Currently you're not checking for instance how many permissions get returned when you searched for `mail`. You'll need to check that the results match the expectations.